src/python/pants/backend/python/tasks/python_execution_task_base.py

```diff
@@ -127,7 +134,16 @@ def create_pex(self, pex_info=None):
           # in the target set's dependency closure.
           pexes = [extra_requirements_pex] + pexes
 
-        constraints = {str(interpreter.identity.requirement)}
+        if pin_selected_interpreter:
+          constraints = {str(interpreter.identity.requirement)}
+        else:
+          constraints = {
+            constraint for rt in relevant_targets if is_python_target(rt)
+          for constraint in PythonSetup.global_instance().compatibility_or_constraints(rt)
```

Comment:
I'm not sure if this works with python ws rules - probably does - but worth fixing the indent anyhow. I suspect you didn't intend this.

Reply:
That's correct, fixing! Thanks!